integration-tests/jail/disk/jail_test.py

```diff
@@ -0,0 +1,170 @@
+        }, addr],
+        "id": "staking_state"
+    }
+    data = json.dumps(q)
+    response = requests.post(client_rpc, headers=headers, data=data)
+    return response.json()["result"]
+
+
+
+
+def create_staking_address(name, passphrase):
+    q = {
+        "method": "wallet_createStakingAddress",
+        "jsonrpc": "2.0",
+        "params": [{
+            "name": name,
+            "passphrase": passphrase
+        }],
+        "id": "wallet_createStakingAddress"
+    }
+    data = json.dumps(q)
+    response = requests.post(client_rpc, headers=headers, data=data)
+
+
+def restore_wallet(name, passphrase, mnemonics):
+    q = {
+        "method": "wallet_restore",
+        "jsonrpc": "2.0",
+        "params": [{
+            "name": name,
+            "passphrase": passphrase
+        }, mnemonics],
+        "id": "wallet_restore_hd"
+    }
+    data = json.dumps(q)
+    response = requests.post(client_rpc, headers=headers, data=data)
+    print("restore wallet {}".format(name), response.json())
+
+def restore_wallets():
+    restore_wallet(
+        "a", "1",
+        "speed tortoise kiwi forward extend baby acoustic foil coach castle ship purchase unlock base hip erode tag keen present vibrant oyster cotton write fetch"
+    )
+
+
+def create_addresses():
+    create_staking_address("a", "1")
+    create_staking_address("a", "1")
+
+
+
+def unjail(name, passphrase, address):
+    q = {
+        "method": "staking_unjail",
+        "jsonrpc": "2.0",
+        "params": [{
+            "name": name,
+            "passphrase": passphrase
+        }, address],
+        "id": "staking_unjail"
+    }
+    data = json.dumps(q)
+    response = requests.post(client_rpc, headers=headers, data=data)
+    print(response.json())
+    return response.json()
+
+
+def check_validators() :
+	try: 
+		x= requests.get('{}/validators'.format(server))
+		data =len(x.json()["result"]["validators"])
+		return data
+	except requests.ConnectionError:
+ 		return 0
+	except:
+		assert False
+
+def wait_for_ready(count) :
+	while True:
+		validators=check_validators()
+		print("{}  current validators={}  waiting for validators={}".format(datetime.datetime.now(),validators, count))
+		if count== validators :
+			print("validators ready")
+			break
+		time.sleep(60)
+
+
+def test_jailing() :
+    print("test jailing")
+    wait_for_ready(2)
+    containers=get_containers()
+    print(containers)
+    if "jail_chain1_1" in containers :
+        assert True
+    else :
+        assert False
+    print("wait for jailing")
+    time.sleep(10)
+    jailthis = containers["jail_chain1_1"]
+    print("jail = " , jailthis)
+    jailthis.kill()
+    wait_for_ready(1)
+    #jailed
+    containers=get_containers()
+    print(containers)
+    if "jail_chain1_1" in containers :
+        assert False
+    else :
+        assert True 
+    print("jail test success")
+
+
+def test_unjailing() :
+    print("test unjailing")
+    wait_for_ready(1)
+
+    count=2
+    while True:
```

Comment:
so this will hang forever (there's no timeout / max attempts) if the test fails?

Reply:
i'll add time limit, maybe two hours.

Reply:
i'll add timeout , if it didn't finish within the time, it will fail

Reply:
Right now the jailing period is 600s, do we want to reduce it to facilitate the test? Moreover, two hours timeout seems to be too long?

p.s. I think Drone.io has a 1 hour timeout right now.

Reply:
i'll change to 1 hour timeout

Reply:
it's configured max 1 hour

Reply:
why not around 10-20 minutes, if the jailing period is 10 minutes in the test?

Reply:
ok